### PR TITLE
Added jobs docs

### PR DESCRIPTION
**Pull request by soltysh**
@pweil- @adellape ptal

### CODE REVIEW

File: architecture/core_concepts/jobs.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Jobs
+
+A job, in contrast to link:deployments.html#replication-controllers[Rreplication Controller],
```
Comment (bfallonf) [inline architecture/core_concepts/jobs.adoc]: s/Rreplication/Replication

**Comment by bfallonf** [comment]
@soltysh I started to suggest some typo fixes, but figured I can pick them up after. I have a few questions, though that you might be able to answer before it's merged:

1. Reading this and the Kubernetes docs, I can understand what a job is and what it does, but I'm not really understanding how this works in the OpenShift context. Can we say how this example helps an OpenShift instance, or maybe get a more helpful example in there?

2. We're adding pretty much the same info to two different locations. Could there just be a sentence of para about jobs in say the architecture/core_concepts/pods_and_services doc that links to the dev_guide/jobs doc that you're made above? 

3. In the scaling section, it mentions the parallelism parameter, but configures the replicas. From what I can see, the parallelism parameter isn't mentioned in the docs in the context of replicas. Wuld it be easier just to say "to scale the job, scale the replicas like normal"?

Thanks!

**Comment by soltysh** [comment]
@bfallonf fixed that typo and here are the answers for your questions:

1. This is meant to be the first step to having scheduled jobs, for now I can't see any useful example than what we have already, or what upstream (k8s) is working on right now. I do believe once we have scheduled jobs we'll extend that part.

2. I've merged into one single doc in dev_guide. In the future, though, I'd like to see a docs explaining replication controllers along with jobs (the provide similar functionality, rc - for ever running pods, jobs - one-time pods) and then pointing to those from deployments (rely on rc) and jobs. But we can leave it for afterwards, I guess. Or you can do that, since you'll better at writing than I am :wink: 

3. The problem is the --replicas parameter in oc scale command, which was introduces before jobs were created. Jobs do not have anything to do with replicas, they have parallelism, which user can control over oc scale. 

I hope that helps.

**Comment by bfallonf** [comment]
@soltysh Cool. What you've got here looks good to me. I think we can merge this as is and I can flesh it out with the following:

1. Good plan. We can put in a meaningful example when we have one. 

2. There's a little on replication controllers atm [1]. I can add some info to this about jobs, then have the info in this PR point to that. WDYT?

3. To clarify, the reader used the --replicas option with the oc scale command to scale both replication controllers and jobs? And this modifies the parallelism parameter in the pod config file? Which, in the case of jobs, determines how many pods will be run until the job will complete?

Bonus question:
4. As this is a new object, does "job" need to be added to the object type table in the CLI doc?

Thanks! As I said, I don't mind adding in the extra stuff. 

[1] https://docs.openshift.com/enterprise/3.0/architecture/core_concepts/deployments.html#replication-controllers
[2] https://docs.openshift.com/enterprise/3.0/cli_reference/basic_cli_operations.html#object-types

**Comment by soltysh** [comment]
@bfallonf 

1. :+1: 

2. Right, but then it would make sense to have both rc and jobs have together. They are meant to supplement each other: run all the time vs run once pods. We can start with just the little info and built down the road on top of that. Totally agree, when we have that element, both deployments and jobs will point to that new document.

3. oc scale --replicas for jobs will change parallelism parameter of a job, for rc it will change replicas parameter of a rc, in short. Generally speaking when invoking scale you need to provide a resource which you want to modify and in result the number of pods the controller (either job or rc) manages will change accordingly. 

4. Yes it should.

File: dev_guide/jobs.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
+= Builds
```
Comment (adellape) [inline dev_guide/jobs.adoc]: Title should be updated to "Jobs" or something similar.

**Comment by adellape** [comment]
Thanks @soltysh. Gonna merge this as-is for @bfallonf to follow up.

**Comment by vikram-redhat** [comment]
@bfallonf followed up with https://github.com/openshift/openshift-docs/pull/1165. So removing the to_followup.

**Pull request merged**